shapeways/oauth2_client.py

```diff
@@ -51,18 +75,57 @@ def authenticate(self, client_id, client_secret):
     def _validate_response(self, response):
         """
         Internal function - validate results
-        :rtype: list()
-        """
-        if response.status_code != 200:
-            raise RuntimeError("Call threw status {}".format(response.status_code))
+        :rtype dict()
+        """
+
+        # Default values - these will be modified as needed by the rate limiting error handling
+        rate_limit = RateLimitingHeaders(
+            rate_limit_type=SW_RATE_LIMIT,
+            is_rate_limited=False
+        )
+        headers = response.headers
+
+        # Handle HTTP errors
+        if response.status_code != requests.codes.ok:
+            if response.status_code == requests.codes.too_many_requests:
+                # We're rate limited
+                rate_limit.is_rate_limited=True
+
+                if CF_RATE_LIMIT_RETRY in headers.keys():
+                    # Limited by CF - backoff for a number of seconds
+                    rate_limit.rate_limit_type=CF_RATE_LIMIT
+                    rate_limit.rate_limit_remaining=0
+                    rate_limit.rate_limit_retry_inseconds=headers[CF_RATE_LIMIT_RETRY]
+                else:
+                    # Shapeways Rate limiting - stupidly, we move the retryInSeconds entry from the response headers
+                    # to the body.  Dealing with this here.
```

Comment:
knowing that we will add it to the headers too, can we handle both cases here ?
so look for and use the headers values, and if not found fall back to the body

Reply:
No - I'll make the change to the client when we actually change the API, not because we want to change the API.  This is how it works today.

Reply:
:( ok